### DISCUSSION TITLE
Roof Removal area override for Port Piscarilius

### DISCUSSION THREAD

**Discussion by superkalaja** [runelite/runelite#18504] [Ideas]
I've recently done some stealing artifacts and while the roof removal plugin works wonders it doesn't function very well within port piscarilius especially when trying to click far away.

I went through the PR's which included the POH custom override and was wondering how much an effort would it be to create a settings toggle for this area.

Of course the workaround currently is just to disable the plugin during stealing artefacts, but I keep forgetting to turn the plugin back on and this sort of an area override for port piscarilius would be nice improvement.

**Comment by Nightfirecat** [comment]
Can you explain a bit further how it "doesn't function very well within port piscarilius"?

**Comment by superkalaja** [comment]
Yes I could've been more elaborate. Typically it's these two buildings (but also the building near the mess hall causes issues. Which don't disappear as they are not supposed to.
![piscarilius_culprits](https://github.com/user-attachments/assets/997f7a6a-ef58-4c40-acc5-8763a40eaf5b)

Switching to different angle to get better view barely helps.
![still_hard_to_click](https://github.com/user-attachments/assets/1f70f5cd-271c-4f25-90fd-9d5fdb491bc6)

If you brave to click through the building, it sometimes does the "you hit roof, so target tile is floor tile of that tile" behaviour and you get routed inside the building (grey tile indicator).
![clicking_through_goes_to_building](https://github.com/user-attachments/assets/5a0e6b0f-ffe6-486a-b54b-d27e3cc23aaa)

Now with roofs removed you can click the blue spot without issues.
![typical_click_target_without_roofs](https://github.com/user-attachments/assets/baf41fe1-c86c-4b03-88ae-84fd21582524)

I guess that another option that hide roofs between cursor and player could be added, but in piscarilius sometimes the roofs of the buildings get in the way as shown in the 3rd screenshot.

I hope this explains it better.

**Comment by superkalaja** [comment]
Notoriously from the south lure, without roof removal you can easily click right.
![from_the_south_lure](https://github.com/user-attachments/assets/d559fcff-2354-4ca7-b168-f731ec845249)

With roof removal it's very difficult to click accurately.
![from_the_south_lure_with_roof_removal](https://github.com/user-attachments/assets/21c0d2b5-ea1e-4734-9126-a20c65404b96)

**Comment by Nightfirecat** [comment]
I think roof removal is working as expected here--roofs are only removed for buildings where you're hovering over tiles which land inside (or very near) the base of the building. (see attached video)

There's not currently a way to add overrides or to hide buildings when you hover a tile which would intersect a building. While I'd be open to eventually working on the latter option, it's not likely to be added soon. I recommend using ground markers to help click through those if you don't want to disable the plugin when in that area.

https://github.com/user-attachments/assets/b7fd718f-3458-492a-900b-a066ef5d0f1d
